base/applications/rapps/misc.cpp

```diff
@@ -39,6 +39,31 @@ CopyTextToClipboard(LPCWSTR lpszText)
     CloseClipboard();
 }
 
+static INT_PTR CALLBACK
+NothingDlgProc(HWND hDlg, UINT uMsg, WPARAM, LPARAM)
+{
+    return uMsg == WM_CLOSE ? DestroyWindow(hDlg) : FALSE;
+}
+
+VOID
+EmulateDialogReposition(HWND hwnd)
```

Comment:
Thinking again about this function, I wonder why you don't call [`CWindow::CenterWindow()`](https://git.reactos.org/?p=reactos.git;a=blob;f=sdk/lib/atl/atlwin.h;hb=058a67cb4a7eef2858536528c1659eadd23d0b6a#l315) in the places where you send the emulate_reposition dialog (since you call that from CMainWindow functions, a class that inherit from CWindow).

(Note that we also have "pure Win32" `CenterWindow()` functions, or `ClipOrCenterWindowToMonitor()`, in our code base already.)

Reply:
This does not center the window. It makes sure the window is in an acceptable position (not off screen if secondary monitor disconnected or resolution changed). The window manager determines where this is.

Reply:
ok makes sense.